### PR TITLE
pre-release cleanup

### PR DESCRIPTION
**Pull request by wjhsf**
* Added comment explaining usage of node 16
* Bumped deps
* Removed unused `.travis.yml`
* Removed unused docker files (99% sure they're unused - @awaterma can you confirm?)
* Noticed in our safe `toString` we check for `null` _then_ check `Array.isArray`, which is unnecessary. Fixed that.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,34 +108,35 @@
     "prettier": "prettier '**/*.{json,ts,yaml,md}'",
     "format": "npm run eslint -- --fix"
   },
+  "//": "We only support node 18+, but v16 still works. We won't block v16 until it becomes a burden.",
```
[the remark this answers is not in the document]
Comment (wjhsf) [reply package.json:111]: We use it a lot 🤷
https://github.com/salesforce/lwc/blob/0c285f4578cf1ccd68e02143f724f11adf27c76e/package.json#L33

[the remark this answers is not in the document]
Comment (wjhsf) [reply package.json:111]: v16 support is kinda secret, so if we want to remove this comment I'd just not mention it anywhere.

[the remark this answers is not in the document]
Comment (colincasey) [reply package.json:111]: But have you seen that used before outside of a Salesforce project? Either way, it's a minor detail that I won't hold up this PR for 😅

**Comment by colincasey** [comment]
I'm in approval of these changes though I believe I've seen @awaterma use the docker-related files as part of his workflow. Let's wait for his review to confirm.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (wjhsf) [reply package.json:111]: I haven't seen it elsewhere in the wild, but it does turn up in a [search](https://github.com/search?q=%22%5C%22%2F%2F%5C%22%22+path%3A%2F%28%5E%7C%5C%2F%29package%5C.json%24%2F&type=code). 🤔

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
   "compilerOptions": {
     /* Language and Environment */
+    // We only support node 18+, but v16 still works. We'll target v16 until it becomes a burden.
```
Comment (awaterma) [inline tsconfig.json:4]: Are these comments legal?

Comment (colincasey) [reply tsconfig.json:4]: @awaterma yes, `tsconfig.json` is [processed as a jsonc file](https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html#tsconfig) (json w/ comments) so this is legal.

**Review by colincasey — APPROVED** [review]

**Pull request merged**